tests/unittests/tests-fmt/tests-fmt.c

```diff
@@ -61,6 +103,39 @@ static void test_fmt_bytes_hex_reverse(void)
     TEST_ASSERT_EQUAL_STRING("06070809", (char *) out);
 }
 
+static void test_fmt_hex_bytes(void)
+{
+    char hex2[2] = "00";
+    uint8_t val1[1] = { 0 };
+    fmt_hex_bytes(val1, hex2);
+    TEST_ASSERT_EQUAL_INT(0, val1[0]);
+
+    memcpy(hex2, "AB", 2);
+    hex2[2] = '\0';
+    val1[0] = 0;
+    fmt_hex_bytes(val1, hex2);
+    TEST_ASSERT_EQUAL_INT(0xAB, val1[0]);
```

Comment:
Here it would also be nice to test what happens if you hand over an empty string ;-).

Reply:
And with strings "strlen(str) % 2 != 0" ;-)